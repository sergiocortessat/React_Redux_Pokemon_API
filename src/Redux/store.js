import { configureStore } from '@reduxjs/toolkit';
import pokemonReducer from './Slices/homeFetchSlice';

export default configureStore({
  reducer: {
    pokemon: pokemonReducer,
  },
});
