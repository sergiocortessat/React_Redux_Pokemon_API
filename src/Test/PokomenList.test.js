import React from 'react';
import { BrowserRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import List from '../App/Containers/PokomenList';
import store from '../Redux/store';

describe('Pokemon List', () => {
  test('should match with snapshot', () => {
    const tree = renderer
      .create(
        <BrowserRouter>
          <Provider store={store}>
            <List />
          </Provider>
        </BrowserRouter>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
