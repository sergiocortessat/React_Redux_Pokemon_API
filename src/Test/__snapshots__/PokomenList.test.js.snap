// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header with Link should match with snapshot 1`] = `
<div>
  <div
    className="MuiFormControl-root MuiTextField-root textField"
    onClick={[Function]}
  >
    <label
      className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-filled"
      data-shrink={false}
    >
      Pokemon Name: 
    </label>
    <div
      className="MuiInputBase-root MuiFilledInput-root MuiFilledInput-underline MuiInputBase-formControl"
      onClick={[Function]}
    >
      <input
        aria-invalid={false}
        autoFocus={false}
        className="MuiInputBase-input MuiFilledInput-input"
        disabled={false}
        onAnimationStart={[Function]}
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        required={false}
        type="text"
        value=""
      />
    </div>
  </div>
  <div
    className="types-filter"
    data-testid="filter"
  >
    <button
      className="type-filter-item"
      data-testid="button"
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "#94bc4a",
          "color": "white",
        }
      }
      type="button"
    >
      Bug
       
    </button>
    <button
      className="type-filter-item"
      data-testid="button"
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "#736c75",
          "color": "white",
        }
      }
      type="button"
    >
      Dark
       
    </button>
    <button
      className="type-filter-item"
      data-testid="button"
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "#6a7baf",
          "color": "white",
        }
      }
      type="button"
    >
      Dragon
       
    </button>
    <button
      className="type-filter-item"
      data-testid="button"
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "#e5c531",
          "color": "white",
        }
      }
      type="button"
    >
      Electric
       
    </button>
    <button
      className="type-filter-item"
      data-testid="button"
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "#e397d1",
          "color": "white",
        }
      }
      type="button"
    >
      Fairy
       
    </button>
    <button
      className="type-filter-item"
      data-testid="button"
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "#cb5f48",
          "color": "white",
        }
      }
      type="button"
    >
      Fighting
       
    </button>
    <button
      className="type-filter-item"
      data-testid="button"
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "#ea7a3c",
          "color": "white",
        }
      }
      type="button"
    >
      Fire
       
    </button>
    <button
      className="type-filter-item"
      data-testid="button"
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "#7da6de",
          "color": "white",
        }
      }
      type="button"
    >
      Flying
       
    </button>
    <button
      className="type-filter-item"
      data-testid="button"
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "#846ab6",
          "color": "white",
        }
      }
      type="button"
    >
      Ghost
       
    </button>
    <button
      className="type-filter-item"
      data-testid="button"
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "#71c558",
          "color": "white",
        }
      }
      type="button"
    >
      Grass
       
    </button>
    <button
      className="type-filter-item"
      data-testid="button"
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "#cc9f4f",
          "color": "white",
        }
      }
      type="button"
    >
      Ground
       
    </button>
    <button
      className="type-filter-item"
      data-testid="button"
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "#70cbd4",
          "color": "white",
        }
      }
      type="button"
    >
      Ice
       
    </button>
    <button
      className="type-filter-item"
      data-testid="button"
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "#aab09f",
          "color": "black",
        }
      }
      type="button"
    >
      Normal
       
    </button>
    <button
      className="type-filter-item"
      data-testid="button"
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "#b468b7",
          "color": "white",
        }
      }
      type="button"
    >
      Poison
       
    </button>
    <button
      className="type-filter-item"
      data-testid="button"
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "#e5709b",
          "color": "white",
        }
      }
      type="button"
    >
      Psychic
       
    </button>
    <button
      className="type-filter-item"
      data-testid="button"
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "#b2a061",
          "color": "white",
        }
      }
      type="button"
    >
      Rock
       
    </button>
    <button
      className="type-filter-item"
      data-testid="button"
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "#89a1b0",
          "color": "white",
        }
      }
      type="button"
    >
      Steel
       
    </button>
    <button
      className="type-filter-item"
      data-testid="button"
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "#539ae2",
          "color": "white",
        }
      }
      type="button"
    >
      Water
       
    </button>
    <button
      className="type-filter-item"
      data-testid="button"
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "red",
          "color": "black",
        }
      }
      type="button"
    >
      Reset
       
    </button>
  </div>
  <div
    className="pokemon-list-items"
  />
</div>
`;

exports[`Pokemon List should match with snapshot 1`] = `
<div>
  <div
    className="MuiFormControl-root MuiTextField-root textField"
    onClick={[Function]}
  >
    <label
      className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-filled"
      data-shrink={false}
    >
      Pokemon Name: 
    </label>
    <div
      className="MuiInputBase-root MuiFilledInput-root MuiFilledInput-underline MuiInputBase-formControl"
      onClick={[Function]}
    >
      <input
        aria-invalid={false}
        autoFocus={false}
        className="MuiInputBase-input MuiFilledInput-input"
        disabled={false}
        onAnimationStart={[Function]}
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        required={false}
        type="text"
        value=""
      />
    </div>
  </div>
  <div
    className="types-filter"
    data-testid="filter"
  >
    <button
      className="type-filter-item"
      data-testid="button"
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "#94bc4a",
          "color": "white",
        }
      }
      type="button"
    >
      Bug
       
    </button>
    <button
      className="type-filter-item"
      data-testid="button"
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "#736c75",
          "color": "white",
        }
      }
      type="button"
    >
      Dark
       
    </button>
    <button
      className="type-filter-item"
      data-testid="button"
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "#6a7baf",
          "color": "white",
        }
      }
      type="button"
    >
      Dragon
       
    </button>
    <button
      className="type-filter-item"
      data-testid="button"
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "#e5c531",
          "color": "white",
        }
      }
      type="button"
    >
      Electric
       
    </button>
    <button
      className="type-filter-item"
      data-testid="button"
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "#e397d1",
          "color": "white",
        }
      }
      type="button"
    >
      Fairy
       
    </button>
    <button
      className="type-filter-item"
      data-testid="button"
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "#cb5f48",
          "color": "white",
        }
      }
      type="button"
    >
      Fighting
       
    </button>
    <button
      className="type-filter-item"
      data-testid="button"
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "#ea7a3c",
          "color": "white",
        }
      }
      type="button"
    >
      Fire
       
    </button>
    <button
      className="type-filter-item"
      data-testid="button"
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "#7da6de",
          "color": "white",
        }
      }
      type="button"
    >
      Flying
       
    </button>
    <button
      className="type-filter-item"
      data-testid="button"
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "#846ab6",
          "color": "white",
        }
      }
      type="button"
    >
      Ghost
       
    </button>
    <button
      className="type-filter-item"
      data-testid="button"
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "#71c558",
          "color": "white",
        }
      }
      type="button"
    >
      Grass
       
    </button>
    <button
      className="type-filter-item"
      data-testid="button"
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "#cc9f4f",
          "color": "white",
        }
      }
      type="button"
    >
      Ground
       
    </button>
    <button
      className="type-filter-item"
      data-testid="button"
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "#70cbd4",
          "color": "white",
        }
      }
      type="button"
    >
      Ice
       
    </button>
    <button
      className="type-filter-item"
      data-testid="button"
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "#aab09f",
          "color": "black",
        }
      }
      type="button"
    >
      Normal
       
    </button>
    <button
      className="type-filter-item"
      data-testid="button"
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "#b468b7",
          "color": "white",
        }
      }
      type="button"
    >
      Poison
       
    </button>
    <button
      className="type-filter-item"
      data-testid="button"
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "#e5709b",
          "color": "white",
        }
      }
      type="button"
    >
      Psychic
       
    </button>
    <button
      className="type-filter-item"
      data-testid="button"
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "#b2a061",
          "color": "white",
        }
      }
      type="button"
    >
      Rock
       
    </button>
    <button
      className="type-filter-item"
      data-testid="button"
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "#89a1b0",
          "color": "white",
        }
      }
      type="button"
    >
      Steel
       
    </button>
    <button
      className="type-filter-item"
      data-testid="button"
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "#539ae2",
          "color": "white",
        }
      }
      type="button"
    >
      Water
       
    </button>
    <button
      className="type-filter-item"
      data-testid="button"
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "red",
          "color": "black",
        }
      }
      type="button"
    >
      Reset
       
    </button>
  </div>
  <div
    className="pokemon-list-items"
  />
</div>
`;
